src/bin/add/main.rs

```diff
@@ -70,20 +72,46 @@ crates.io registry suggests. One goal of `cargo add` is to prevent you from usin
 dependencies (version set to "*").
 "#;
 
+fn print_msg(dep: &Dependency, section: &[String], optional: bool) {
+    let optional = if optional { "optional " } else { "" };
+    let section = if section.len() == 1 {
+        section[0].clone()
+    } else {
+        format!("{} for target {}", &section[2], &section[1])
+    };
+
+    let mut output = StandardStream::stdout(ColorChoice::Auto);
+    output
+        .set_color(ColorSpec::new().set_fg(Some(Color::Green)).set_bold(true))
+        .unwrap();
+    print!("{:>12}", "Adding");
+    output.reset().unwrap();
+    println!(
+        " {} v{} to {}{}",
+        dep.name,
+        dep.version().unwrap_or(""),
+        optional,
+        section
+    );
+}
+
 fn handle_add(args: &Args) -> Result<()> {
     let manifest_path = args.flag_manifest_path.as_ref().map(From::from);
     let mut manifest = Manifest::open(&manifest_path)?;
     let deps = &args.parse_dependencies()?;
 
     deps.iter()
         .map(|dep| {
+            if !args.flag_quiet {
+                print_msg(dep, &args.get_section(), args.flag_optional);
```

Comment:
Once this returns a Result we can decide if we want to _ignore_ it here: Failing to print this should not crash cargo-add IMO

Reply:
Good catch! Maybe we should check for tty like cargo [does](https://github.com/rust-lang/cargo/blob/2316ca0b7d6b756a4fb85f3a32a7c12bf6e162c9/src/cargo/core/shell.rs#L280)?

Reply:
Yeah, probably for the best.

Reply:
So it seems like cargo has additional logic for determining when to print color. Termcolor just looks at the $TERM environment variable, while cargo also calls isatty. The latter is definitely preferable, as in the case @killercup mentioned above (piping to a log file), the $TERM variable will be set.

Should I just copy that code from cargo? I'm beginning to think it might be worth it to do `extern crate cargo` and use its functionality for printing messages directly. It'd have the additional benefit that if cargo ever changes its output format, we don't have to do anything to keep up.